dist/configs/itwinjs-recommended.js

```diff
@@ -170,12 +170,14 @@ module.exports =
       }
     ],
     "@typescript-eslint/return-await": "error",
+    "@typescript-eslint/no-duplicate-type-constituents": "off",
     "@typescript-eslint/no-this-alias": "error",
     "@typescript-eslint/no-use-before-define": "off",
     "@typescript-eslint/no-unnecessary-type-assertion": "error",
     "@typescript-eslint/no-unsafe-argument": "off",
     "@typescript-eslint/no-unsafe-assignment": "off",
     "@typescript-eslint/no-unsafe-call": "off",
+    "@typescript-eslint/no-unsafe-enum-comparison": "off",
```

Comment:
I think `no-unsafe-enum-comparison` is actually useful. The whole point of having an enum is that you don't have to use magic numbers/strings, and that the underlying values could change without affecting the rest of your code. Disabling this rule allows people to sneak in stinky code that directly use enum values as "magic numbers" instead of using the enum object. I don't like that.

Reply:
I certainly agree that this should generally be considered a code smell, but IMO this rule breaks way too many perfectly appropriate patterns, including:

- Exhaustive switch statements to check untyped input (e.g., [lookupCategory](https://github.com/iTwin/itwinjs-core/blob/7b693b7a87d0ed2e54377877612a7921fb26655c/core/bentley/src/StatusCategory.ts#L137))
- Bitwise operations on bitmask/"flags" enums (e.g., [ClipShape.createBlock](https://github.com/iTwin/itwinjs-core/blob/7b693b7a87d0ed2e54377877612a7921fb26655c/core/geometry/src/clipping/ClipPrimitive.ts#L558-L559))
- Using an enum as just a non-exhaustive set of constants (e.g., [HSVConstants](https://github.com/iTwin/itwinjs-core/blob/7b693b7a87d0ed2e54377877612a7921fb26655c/core/common/src/HSVColor.ts#L15-L19))

That last one is debatable, but the fact is we do these things _everywhere_ in itwinjs-core.  Enabling this spat out **805** additional lint errors, and after looking thru the first 100 or so I couldn't find any places where I didn't just want to add an ignore comment, so IMHO this was creating more noise than anything else.

I think a version that let us only ban comparisons where one side was a literal would be a much more targeted and useful rule.

Reply:
A possible middle ground could be disabling the rule for all of core repo by adding this rule here: https://github.com/iTwin/itwinjs-core/blob/master/common/config/eslint/eslint.config.base.js

Not all pkgs import this tho, so you might have to manually import for some pkgs like so: https://github.com/iTwin/itwinjs-core/blob/master/core/backend/eslint.config.js

This way we can have enforce this for all other repos consuming recommended rules while core repo can be exempt and no need to manually go in and change a bunch of stuff that exists there

Reply:
Just sharing my 2 cents on the patterns you mentioned:
- I think `lookupCategory` _is_ a code smell. At the very least, `BentleyError.errorNumber` should be a narrower type than just `number`.
- Bitmasks shouldn't be a problem if you use enum types:
https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=KYOwrgtgBAKgngB2FA3gKClAYgJQKoCSMUAvFAAwAeAjADQZQBqAogOKsHOkWUBM9AXzRoAxgHsQAZwAuUaQAsAliADmALliJkZeEgB0uQsQA%2Bm-S3acA3GiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZonpNYAW1pkOY4nyrIOTdFETRoHaJHBgAviB1A&tsconfig=&tokens=false
- Why is _that_ an enum????
```ts
export namespace HSVConstants {
  const VISIBILITY_GOAL = 40;
  const HSV_SATURATION_WEIGHT = 4;
  const HSV_VALUE_WEIGHT = 2;
}
```

IMHO it all boils down to widespread smelly usage of enums in iTwin.js :/

Reply:
I agree with Anmol's suggestion, if it's too hard to clean up core, we can disable the rule there, but there's no reason to disable it for everyone. It's a good rule, and we should discourage people from writing smelly code even if we have some in core.

Reply:
On `lookupCategory`, I agree that was probably not the best example, but there are a lot of places (deserializing JSON for example) where all you (should) have is `unknown`, and I would much rather use comparisons to figure out what that is.  Today this rule strongly encourages you to just cast instead, which IMO is way worse.

Bitmasks are still a problem - remember this rule only applies to _comparisons_: https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=KYOwrgtgBAKgngB2FA3gKClAYgJQKoCSMUAvFAAwAeAjADQZQBqAogOKsHOkWUBM9AXzRoAxgHsQAZwAuUAEYBDEEoXd4SAHS5CMURJlQAlpIBCK5VgBOYQ9IBc8sWIA2wJWsTAt%2BIqRJkACnUvbV8AMigAM2tbAEoAbiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZonpNYAW1pkOY4nyrIOTdFETRoHaJHBgAviB1A&tsconfig=&tokens=false

And I agree about the last one being gross - I think a lot of that comes from C++ where enums really are little more that organized constants.

I'm fine w/ disabling in core instead, I'm just not sure I agree this is that good of a rule - to me it does as much harm as good.

Reply:
I would suggest this pattern instead: https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=KYOwrgtgBAKgngB2FA3gKClAYgJQKoCSMUAvFAAwAeAjADQZQBqAogOKsHOkWUBM9AXzRoAxgHsQAZwAuUAEYBDEEoXd4SAHS5CMURJlQAlpIBCK5VgBOYQ9IBc8sWIA2wJdwAU64FvxEoAGTy5goAlFAAhGTkANxAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZonpNYAW1pkOY4nyrIOTdFETRoHaJHBgAviB1A&tsconfig=&tokens=false
I'm not a fan of repeating the enum member twice anyway. But that's a personal preference and I get where you're coming from. The rule does require changing some patterns that are innocent, but it also prevents lots of code smells so I think it's still a net positive. Let's just disable it in core for now, and if other people complain about this we can come back to disabling it here.

Reply:
does that even exist in JS?

Reply:
https://github.com/iTwin/itwinjs-core/compare/as/bump-eslint-plugin?diff=unified&w=#diff-8b7494799d85368c26b3bb81c75e0d542a3a48a6b2813ef94c45f33872efbe79R433

Oh ya

Reply:
I'm gonna revert this here and apply it in core per your suggestions, because this is starting to veer into bikeshedding...

Reply:
Sounds good, TIL new word 😄